### ISSUE TITLE
Add samba server support (through kio? fuse?)

### ISSUE THREAD

**Issue by HinzundKunz** [spersson/Kup#13]
It would be nice to be able to do a backup or sync to a samba server. Right now the only software that can do this is sbackup, which is gnome.

**Comment by Silicomancer** [comment]
For me this would be extremely useful too. Do you still consider implementing this?

**Comment by carlonluca** [comment]
I do this by mounting via cifs the smb share. Works perfectly. Some more optimizations may improve the user experience though.

**Comment by spersson** [comment]
@carlonluca: yes, it seems you are right. There's no need for a fuse mount, from reading around a bit now it seems linux kernel mount is possible as long as cifs-utils is installed. It should be possible to check if user's system has mount.cifs installed with SUID root permissions and if so, offer the user an easier way to configure backup to a cifs share. As a first step just asking for server name and file path should be OK, later would be nice to add server browsing. 
I assume username/password should be stored in kwallet.
Then the next problem is how to detect if the samba share is available. What I can think of is to either wait for the user to mount the share somewhere, I think this can be detected using the solid library. Or the other option is to regularly, once every 5 minutes or so, call nmblookup (netbios name lookup) to get the IP address of the samba server and then try pinging it. If ping is successful it can be considered available until ping fails (keep pinging every few minutes). If user decides to start a backup job, only then we do the actual mounting of the network share, in some random new folder created to be used as a mounting point.

If any of you guys have other ideas or opinions on how this should be done, please let me know!
Thanks.

**Comment by carlonluca** [comment]
What you say makes sense. I would however also bring into the discussion a couple of other ideas. This https://github.com/KDE/smb4k/blob/master/helpers/smb4kmounthelper.cpp mounts the smb share, taking care of privilege elevation.
Also determining whether smb support is available can be done by looking at /proc/filesystems.
Similar implementation can be probably used also for other protocols.

**Comment by Pointedstick** [comment]
If and when KIO gains native FUSE mount support, this will be transparently possible automatically. You can follow https://bugs.kde.org/show_bug.cgi?id=75324. There's actually a GSGoogle Summer of Code student working on it right now!
